templates/stackimage/Dockerfile

```diff
@@ -0,0 +1,90 @@
+ARG GRADLE_VERSION=6.8.2
+ARG SHA=8de6efc274ab52332a9c820366dd5cf5fc9d35ec7078fd70c8ec6913431ee610
+ARG BASE_URL=https://services.gradle.org/distributions
+ARG GRADLE_DIR=/usr/share/gradle
+
+RUN mkdir -p ${GRADLE_DIR} /tmp/stack/gradle/unzip\
+  && curl -fsSL -o /tmp/stack/gradle/gradle.zip ${BASE_URL}/gradle-${GRADLE_VERSION}-bin.zip \
+  && echo "${SHA}  /tmp/stack/gradle/gradle.zip" | sha256sum --check - \
+  && unzip -d /tmp/stack/gradle/unzip /tmp/stack/gradle/gradle.zip \
+  && cp -r /tmp/stack/gradle/unzip/gradle-*/* ${GRADLE_DIR} \
+  && rm -rf /tmp/stack \
+  && ln -s ${GRADLE_DIR}/bin/gradle /usr/bin/gradle
+
+# 2. Cache all needed data required to build and run the liberty server in dev mode.
+FROM prereq AS builder
+
+ENV OPENJ9_JAVA_OPTIONS="-Xshareclasses:name=liberty,nonfatal,cacheDir=/output/.classCache/"
+
+RUN mkdir -p /stacks/java-openliberty/priming-app \
+  && mkdir -p /stacks/java-openliberty/config \
+  && chown -R java_user /stacks \
+  && mkdir -p /output \
+  && chown -R java_user /output \
+  && chmod 775 /output
+
+COPY ./LICENSE /licenses/
+
+USER java_user
+
+COPY ./config /stacks/java-openliberty/config
+COPY ./priming-app/src /stacks/java-openliberty/priming-app/src
+COPY ./priming-app/build.gradle ./priming-app/settings.gradle /stacks/java-openliberty/priming-app/
+
+WORKDIR /stacks/java-openliberty/priming-app
+
+ARG LIBERTY_RUNTIME_VERSION={{.OL_RUNTIME_VERSION}}
+
+# Build the priming application to populate the cache (java_user home dir: /home/java_user/.gradle).
+RUN umask -S u=rwx,g=rwx,o=rx; gradle build -Dliberty.runtime.version=${LIBERTY_RUNTIME_VERSION} --no-daemon
+
+# Start the Open Liberty server in dev mode. There are dependencies that are detected during dev mode execution due to the use of mojo executor, etc.
+RUN nohup bash -c "umask -S u=rwx,g=rwx,o=rx; gradle libertyDev -Dliberty.runtime.version=${LIBERTY_RUNTIME_VERSION} --hotTests &" \
+ && sleep 600 \
+ && echo  && echo "Done sleeping" && echo
+
+# Delete the defaultServer directory created during libertyDev task execution. Users will customize their own. 
+RUN rm -rf /stacks/java-openliberty/priming-app/build/wlp/usr/servers/defaultServer
+
+# 3. Gather all installed software, Open Liberty basic installation, and cached Open Liberty data into the final image. 
+FROM prereq
+
+ENV OPENJ9_JAVA_OPTIONS="-Xshareclasses:name=liberty,nonfatal,cacheDir=/output/.classCache/"
+
+RUN mkdir -p /output \
+  && chown -R java_user /output \
+  && chmod 775 /output \
+  && mkdir -p /opt/ol/wlp \
+  && chown -R java_user /opt/ol \
+  && chmod 775 /opt/ol /opt/ol/wlp \
+  && mkdir -p /work/outer-loop-app \
+  && chown -R java_user /work \
+  && chmod -R 775 /work \
+  && mkdir -p /stack/ol/config/configDropins/defaults \
+  && chmod -R 775 /stack/ol/config/configDropins/defaults
+
+COPY --chown=1001:0 --from=builder /home/java_user/.gradle /.gradle
+
+# Remove daemon cached data to avoid write issues.
+RUN  rm -rf /.gradle/daemon; chmod -R 775 /.gradle
```

Comment:
So we're only caching the exact same stuff as the Maven stack image?
* the /opt/ol/wlp install
* the M2 repo contents?
We're not caching anything at all with .gradle?  

It seems like if that's the case we could also just use the existing image.    Though there's value in having parallel builds even with repetition..just thinking out loud / asking.

Reply:
The most significant difference here is the gradle vs maven installation. All else should be similar.

Reply:
Can you open an issue then on this repo to investigate looking at using the existing stack image?    Just to remind us of this idea..not to rush and do it.